pkg/services/baremetal/client/ssh/ssh_client.go

```diff
@@ -751,3 +765,47 @@ func removeUselessLinesFromCloudInitOutput(s string) string {
 	}
 	return strings.Join(output, "\n")
 }
+
+func (c *sshClient) ExecutePreProvisionCommand(ctx context.Context, command string) (int, string, error) {
+	client, clientConfig, err := c.getSSHClient()
+	if err != nil {
+		return 0, "", err
+	}
+	defer client.Close()
+
+	scpClient, err := scp.NewClientBySSH(client)
```

Comment:
why do you need this? Why is "normal ssh" not enough here?

Reply:
We copy a binary here. I want to be sure it gets copied correctly.